src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/BlobManager.cs

```diff
@@ -1105,22 +1109,58 @@ await this.PerformWithRetriesAsync(
                         });
                 }
 
-                if (jsonString == null)
-                {
-                    return false;
-                }
-                else
-                {
-                    // read the fields from the json to update the checkpoint info
-                    JsonConvert.PopulateObject(jsonString, this.CheckpointInfo);
-                    return true;
-                }
             }
             catch (Exception e)
             {
-                this.HandleStorageError(nameof(FindCheckpointsAsync), "could not find any checkpoint", checkpointCompletedBlob.Name, e, true, this.PartitionErrorHandler.IsTerminated);
+                this.HandleStorageError(nameof(FindCheckpointsAsync), "could not find any checkpoint", this.checkpointCompletedBlob.Name, e, true, this.PartitionErrorHandler.IsTerminated);
+                throw;
+            }
+
+            if (jsonString == null)
+            {
+                return false;
+            }
+
+            try
+            {
+                // read the fields from the json to update the checkpoint info
+                JsonConvert.PopulateObject(jsonString, this.CheckpointInfo);
+            }
+            catch (JsonException e)
+            {
+                this.HandleStorageError(nameof(FindCheckpointsAsync), "could not parse json file describing last checkpoint", this.checkpointCompletedBlob.Name, e, true, false);
                 throw;
             }
+
+            if (this.CheckpointInfo.RecoveryAttempts > 0 || DateTimeOffset.UtcNow - lastModified > TimeSpan.FromMinutes(5))
+            {
+                this.CheckpointInfo.RecoveryAttempts++;
+
+                this.TraceHelper.FasterProgress($"Incremented recovery attempt counter to {this.CheckpointInfo.RecoveryAttempts} in {this.checkpointCompletedBlob.Name}.");
+
+                await this.WriteCheckpointMetadataAsync();
+
+                if (this.CheckpointInfo.RecoveryAttempts > 3 && this.CheckpointInfo.RecoveryAttempts < 30)
+                {
+                    this.TraceHelper.BoostTracing = true;
+                }
+            }
+
+            return true;
```

Comment:
I don't see why capping the counter itself would be useful. No matter how large, it will still give us useful information (also in the traces). 

Or did you mean to cap the actual recovery attempts?